dt-reports/reports.php

```diff
@@ -310,6 +310,54 @@ public static function get( $value, $type ) : array {
         return $report;
     }
 
+    /**
+     * Get posts filtered by where clauses
+     *
+     * @param array $where An assoc array of key, value pairs
+     */
+    public static function where( $where = [] ) {
+        global $wpdb;
+        $report = [];
+
+        $sql = '';
+        $args = [];
+
+        foreach ( $where as $key => $value ) {
+            $new_sql = '';
+            if ( !empty( $sql ) ) {
+                $new_sql .= ' AND';
+            } else {
+                $new_sql .= ' WHERE';
+            }
+
+            $new_sql .= $wpdb->prepare( ' %1$s = ', $key );
+
+            if ( is_numeric( $value ) ) {
+                $new_sql .= '%d';
+            } else if ( is_string( $value ) ) {
+                $new_sql .= '%s';
+            } else {
+                dt_write_log( __METHOD__ . ' $value is of type ' . gettype( $value ) . ' but should be a string or number' );
```

Comment:
@squigglybob can we return an error instead?

Reply:
sure :)

Reply:
done